### PR TITLE
feat: spam automation

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

This PR does two things:
- Changes the action that auto-closes those spammy .gitignore PRs to use the `spam` label so folks can be disqualified from Hacktoberfest
- Adds a new action that hooks into the label event, and checks if a PR is spam. If so, it comments explaining that the PR is spam and that they should read our contributing guidelines.

### CODE REVIEW

File: .github/workflows/spam.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: GitHub - Spam PR
+on:
+  pull_request_target:
+    types:
+      - labeled
+
+jobs:
+  has-translation:
```
Comment (moT01) [inline .github/workflows/spam.yml:8]:
```suggestion
  is-spam:
```

**Review by moT01 — COMMENTED** [review]
> two or more PRs marked as spam will result in your permanent disqualification

Do we really want to automate this? I don't particularly like that, but I won't put up much of a fight. I feel like we had this conversation last year. If we remove the spam label, does it still count towards that?

**Comment by naomi-lgbt** [comment]
> Do we really want to automate this?

Automate the message? Or the label?

**Comment by moT01** [comment]
Automate the label

**Comment by naomi-lgbt** [comment]
That was true of the `invalid` label last year, when we implemented this automation.

**Comment by naomi-lgbt** [comment]
> If we remove the spam label, does it still count towards that?

If they are already disqualified, removing the label won't turn that off - they'd need to reach out to the Hacktoberfest team to have the account re-enabled.

But if it's their FIRST spam PR, then removing the label makes it not count yes.

**Review by moT01 — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
I am happy with this.

Clarify: Is this meant to remain as is even after October?

**Comment by naomi-lgbt** [comment]
The spam label on the autoclose will stay, but the comment will go away november 1st.

or 13th, given my memory.

**Pull request merged**